Turkey/TestRunner.cs

```diff
@@ -48,6 +48,7 @@ public class TestOutput
         public async virtual Task BeforeTestAsync() {}
         public async virtual Task AfterParsingTestAsync(string name, bool enabled) {}
         public async virtual Task AfterRunningTestAsync(string name, TestResult result, StringBuilder testLog, TimeSpan testTime) {}
+        public async virtual Task PrintFailedTests() {}
```

Comment:
Nit: I prefer the previous implementation (which re-used `AfterRunningAllTestsAsync` for this) since the design would not implicitly recommend each output format to print failed tests. But the output format can still print failed tests after running everything if they want.